src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -0,0 +1,609 @@
+        {
+            return false;
+        }
+
+        JSONHelpers::DeviceInfoJSON deviceInfoJson{ uniqueId, *deviceInfo };
+        JSONHelpers::SerializeDeviceInfoToTmpFile(deviceInfoJson, activeZoneSetTmpFileName);
+
+        return true;
+    }
+
+    void FancyZonesData::ParseDataFromTmpFiles()
+    {
+        ParseDeviceInfoFromTmpFile(activeZoneSetTmpFileName);
+        ParseDeletedCustomZoneSetsFromTmpFile(deletedCustomZoneSetsTmpFileName);
+        ParseCustomZoneSetFromTmpFile(appliedZoneSetTmpFileName);
+        SaveFancyZonesData();
+    }
+
+    void FancyZonesData::ParseDeviceInfoFromTmpFile(std::wstring_view tmpFilePath)
+    {
+        std::scoped_lock lock{ dataLock };
+        const auto& deviceInfo = JSONHelpers::ParseDeviceInfoFromTmpFile(tmpFilePath);
+
+        if (deviceInfo)
+        {
+            deviceInfoMap[deviceInfo->deviceId] = std::move(deviceInfo->data);
+        }
+    }
+
+    void FancyZonesData::ParseCustomZoneSetFromTmpFile(std::wstring_view tmpFilePath)
+    {
+        std::scoped_lock lock{ dataLock };
+        const auto& customZoneSet = JSONHelpers::ParseCustomZoneSetFromTmpFile(tmpFilePath);
+
+        if (customZoneSet)
+        {
+            customZoneSetsMap[customZoneSet->uuid] = std::move(customZoneSet->data);
+        }
+    }
+
+    void FancyZonesData::ParseDeletedCustomZoneSetsFromTmpFile(std::wstring_view tmpFilePath)
+    {
+        std::scoped_lock lock{ dataLock };
+        const auto& deletedCustomZoneSets = JSONHelpers::ParseDeletedCustomZoneSetsFromTmpFile(tmpFilePath);
+        for (const auto& zoneSet : deletedCustomZoneSets)
+        {
+            customZoneSetsMap.erase(zoneSet);
+        }
+    }
+
+    json::JsonObject FancyZonesData::GetPersistFancyZonesJSON()
+    {
+        return JSONHelpers::GetPersistFancyZonesJSON(zonesSettingsFileName, appZoneHistoryFileName);
+    }
+
+    void FancyZonesData::LoadFancyZonesData()
+    {
+        if (!std::filesystem::exists(zonesSettingsFileName))
+        {
+            MigrateCustomZoneSetsFromRegistry();
+
+            SaveFancyZonesData();
+        }
+        else
+        {
+            json::JsonObject fancyZonesDataJSON = GetPersistFancyZonesJSON();
+
+            appZoneHistoryMap = JSONHelpers::ParseAppZoneHistory(fancyZonesDataJSON);
+            deviceInfoMap = JSONHelpers::ParseDeviceInfos(fancyZonesDataJSON);
+            customZoneSetsMap = JSONHelpers::ParseCustomZoneSets(fancyZonesDataJSON);
+        }
+    }
+
+    void FancyZonesData::SaveFancyZonesData() const
+    {
+        std::scoped_lock lock{ dataLock };
+        json::JsonObject root{};
+        json::JsonObject appZoneHistoryRoot{};
+
+        appZoneHistoryRoot.SetNamedValue(L"app-zone-history", JSONHelpers::SerializeAppZoneHistory(appZoneHistoryMap));
+        root.SetNamedValue(L"devices", JSONHelpers::SerializeDeviceInfos(deviceInfoMap));
+        root.SetNamedValue(L"custom-zone-sets", JSONHelpers::SerializeCustomZoneSets(customZoneSetsMap));
```

Comment:
Please replace the const strings with const variables and explicitly mark them as non-localizable.